security/src/main/java/de/evoila/cf/security/credhub/CredhubConnection.java

```diff
@@ -16,35 +21,48 @@
 import org.springframework.security.oauth2.common.AuthenticationScheme;
 import org.springframework.stereotype.Service;
 
+import javax.naming.ConfigurationException;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
 import java.util.Arrays;
 
+
 /**
  * Created by reneschollmeyer, evoila on 15.11.18.
  */
 @Service
+@ConditionalOnBean(CredhubBean.class)
 public class CredhubConnection {
 
     private static final String GRANT_TYPE="client_credentials";
 
     private CredhubBean credhubBean;
 
-    private ClientHttpRequestFactory clientHttpRequestFactory;
+    private KeyStoreHandler keyStoreHandler;
 
-    public CredhubConnection(CredhubBean credhubBean) {
+    public CredhubConnection(CredhubBean credhubBean, KeyStoreHandler keyStoreHandler) {
         this.credhubBean = credhubBean;
+        this.keyStoreHandler = keyStoreHandler;
     }
 
-    @ConditionalOnBean(AcceptSelfSignedClientHttpRequestFactory.class)
-    @Autowired(required = false)
-    private void selfSignedRestTemplate(AcceptSelfSignedClientHttpRequestFactory requestFactory) {
-        clientHttpRequestFactory = requestFactory;
-    }
+    public CredHubTemplate createCredhubTemplate() throws KeyStoreException, NoSuchAlgorithmException, ConfigurationException, UnrecoverableKeyException, KeyManagementException {
+
+        SSLConnectionSocketFactory socketFactory = new SSLConnectionSocketFactory(
+                new SSLContextBuilder()
+                    .loadTrustMaterial(null, new TrustSelfSignedStrategy())
+                    .loadKeyMaterial(keyStoreHandler.getKeyStore(
+                            credhubBean.getCertificate().getCert(),
+                            credhubBean.getCertificate().getPrivateKey(),
+                            credhubBean.getCertificate().getCa(),
+                            credhubBean.getKeystorePassword()), credhubBean.getKeystorePassword().toCharArray())
```

Comment:
What happens if getCertificate() or getKeystorePassword() are empty? For getKeystorePassword() this could be the case, as it is not mandatory

Reply:
I actually made it mandatory as a conditionalOnProperty value for the credhub bean

Reply:
ok, but for keystorePassword it makes no sense, because it is not mandatory

Reply:
Done